business_requirement_deliverable_project/tests/test_br.py

```diff
@@ -0,0 +1,286 @@
+            ],
+        }
+
+        self.brA = self.env['business.requirement'].create(vals)
+        self.brB = self.env['business.requirement'].create(vals)
+        self.brC = self.env['business.requirement'].create(vals)
+
+    def test_br_state(self):
+        # test when state=draft
+        self.brA.state = 'draft'
+        self.brB.state = 'draft'
+        self.brC.state = 'draft'
+        try:
+            action = self.projectA.generate_project_wizard()
+        except Exception, e:
+            action = False
+            self.assertEqual(action, False)
+            if type(e) == osv.except_osv:
+                self.assertEqual(
+                    type(e),
+                    osv.except_osv
+                )
+
+        # test when state=confirmed
+        self.brA.state = 'confirmed'
+        self.brB.state = 'confirmed'
+        self.brC.state = 'confirmed'
+        try:
+            action = self.projectA.generate_project_wizard()
+        except Exception, e:
+            action = False
+            self.assertEqual(action, False)
+            if type(e) == osv.except_osv:
+                self.assertEqual(
+                    type(e),
+                    osv.except_osv
+                )
+
+        # test when state=approved
+        self.brA.state = 'approved'
+        self.brB.state = 'confirmed'
+        self.brC.state = 'draft'
+        try:
+            action = self.projectA.generate_project_wizard()
+        except Exception, e:
+            action = False
+            self.assertEqual(action, False)
+            if type(e) == osv.except_osv:
+                self.assertEqual(
+                    type(e),
+                    osv.except_osv
+                )
+
+        # test when state=approved
+        self.brA.state = 'approved'
+        self.brB.state = 'approved'
+        self.brC.state = 'draft'
+        try:
+            action = self.projectA.generate_project_wizard()
+        except Exception, e:
+            action = False
+            self.assertEqual(action, False)
+            if type(e) == osv.except_osv:
+                self.assertEqual(
+                    type(e),
+                    osv.except_osv
+                )
+
+        # test when state=approved
+        self.brA.state = 'approved'
+        self.brB.state = 'approved'
+        self.brC.state = 'confirmed'
+        try:
+            action = self.projectA.generate_project_wizard()
+        except Exception, e:
+            action = False
+            self.assertEqual(action, False)
+            if type(e) == osv.except_osv:
+                self.assertEqual(
+                    type(e),
+                    osv.except_osv
+                )
+
+        # test when state=approved
+        self.brA.state = 'approved'
+        self.brB.state = 'approved'
+        self.brC.state = 'approved'
+        try:
+            action = self.projectA.generate_project_wizard()
+        except Exception:
+            action = False
+        if action:
```

Comment:
You don't need a double validation for the same variable this is similar to `if action: if not action != False:`